Parallel build fails

Ubuntu 16.04:

```
sudo apt install libxml2-dev libcunit1-dev
git clone https://github.com/bblfsh/libuast
cd libuast
cmake -DCMAKE_BUILD_TYPE=Release .
make -j2
```

I get

```
-- The C compiler identification is GNU 4.8.4
-- Check for working C compiler: /usr/bin/cc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found LibXml2: /usr/lib/x86_64-linux-gnu/libxml2.so (found version "2.9.1") 
-- The CXX compiler identification is GNU 4.8.4
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /home/travis/gopath/src/gopkg.in/src-d/hercules.v2/libuast
Scanning dependencies of target example1
Scanning dependencies of target objlib
[ 14%] Building CXX object examples/CMakeFiles/example1.dir/example1.cpp.o
[ 28%] Building C object CMakeFiles/objlib.dir/src/uast.c.o
[ 42%] Building C object CMakeFiles/objlib.dir/src/roles.c.o
[ 42%] Built target objlib
Scanning dependencies of target testing_exec
[ 57%] Building CXX object tests/CMakeFiles/testing_exec.dir/main.cpp.o
Linking CXX executable example1
/usr/bin/ld: cannot find -luast
collect2: error: ld returned 1 exit status
make[2]: *** [examples/example1] Error 1
make[1]: *** [examples/CMakeFiles/example1.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 71%] Building C object tests/CMakeFiles/testing_exec.dir/__/src/testing-tools.c.o
[ 85%] Building C object tests/CMakeFiles/testing_exec.dir/__/src/uast.c.o
[100%] Building C object tests/CMakeFiles/testing_exec.dir/__/src/roles.c.o
Linking CXX executable testing_exec
[100%] Built target testing_exec
make: *** [all] Error 2
```

I reproduce it in 100% of the cases locally and in Travis CI Trusty (14.04) environment.

`make -j1` always works. I think that some of the dependencies are not set in `CMakeLists.txt`.

--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -7,4 +7,4 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17" CACHE STRING "" FORCE)
 
 add_executable(example example.cc)
 target_link_libraries(example uast)
-
+target_link_libraries(example ${libshared})

